### ISSUE TITLE
Global bib file search does not return all results

### ISSUE THREAD

**Issue by Rothbard-m** [nvim-telescope/telescope-bibtex.nvim#56]
Hello and thank-you for an excellent plug-in

I installed and configured this plug-in this evening but have run into an issue where the results in the telescope search window do not match all of the references in the .bib file. I'm sure it is something simple that I haven't been able to find but I am hoping you can help.

My bib file has approx 20 references but the telescope search returns only 12. To help debug I have included my config code and a version of my bib file that includes two references (shortened to make it easier to look at). When I run `:Telescope bibtex` on this shortened version, telescope returns the "salop1982" but not "byrne2022".

I am using the default config and have defined a global bib file. I am calling the telescope-bibtex extension after the config is loaded.

----------------------------------------------------------------------------------------
Telescope config function
----------------------------------------------------------------------------------------
    config = function()
      local telescope = require('telescope')
      telescope.setup{
        extensions = {
          bibtex = {
            -- Depth for the *.bib file
            depth = 1,
            -- Custom format for citation label
            custom_formats = {},
            -- Format to use for citation label.
            -- Try to match the filetype by default, or use 'plain'
            format = '',
            -- Path to global bibliographies (placed outside of the project)
            global_files = { "~/Documents/bib/eco.bib"},
            -- Define the search keys to use in the picker
            search_keys = { 'author', 'year', 'title' },
            -- Template for the formatted citation
            citation_format = '{{author}} ({{year}}), {{title}}.',
            -- Only use initials for the authors first name
            citation_trim_firstname = true,
            -- Max number of authors to write in the formatted citation
            -- following authors will be replaced by "et al."
            citation_max_auth = 2,
            -- Context awareness disabled by default
            context = false,
            -- Fallback to global/directory .bib files if context not found
            -- This setting has no effect if context = false
            context_fallback = true,
            -- Wrapping in the preview window is disabled by default
            wrap = false,
          },
        },
      }
      telescope.load_extension('luasnip')
      telescope.load_extension('bibtex')
    end,
----------------------------------------------------------------------------------------

----------------------------------------------------------------------------------------
Bib file - ~/Documents/bib/eco.bib
----------------------------------------------------------------------------------------
@article{salop1982,
  title={The theory of sales: A simple model of equilibrium price dispersion with identical agents},
  author={Salop, Steven and Stiglitz, Joseph E},
  journal={The American Economic Review},
  volume={72},
  number={5},
  pages={1121--1130},
  year={1982},
  publisher={JSTOR}
}
@article{byrne2022,
  issn={0033-5533},
  journal={The Quarterly journal of economics},
  pages={2499--2537},
  volume={137},
  number={4},
  year={2022},
  title={Price Discrimination by Negotiation: a Field Experiment in Retail Electricity},
  author={Byrne, David P and Martin, Leslie A and Nah, Jia Sheen}
}
----------------------------------------------------------------------------------------

**Comment by noahares** [comment]
I will have a look. Thanks for providing the config and entries.

**Comment by noahares** [comment]
Sorry for taking so long. I just tried to reproduce the issue, but for me it works as expected. I copied your config, set the global bib file and ran `Telescope bibtex` in a new `tex` file. Are you using latex or markdown?

**Comment by Rothbard-m** [comment]
Hi Noahares - no problem, thanks for taking the time to review my issue.

I am using latex, not markdown. I tried cleaning the install of `Telescope bibtex` and then re-installing with no change.

Another point I forgot to make in my first post is that I'm using Vimtex for managing the compilation of the `tex` file and I find the omni complete menu launched in the context of `\cite{` lists all references I expect to see. I'm not sure if this is useful information in understanding the issue?

I had a look at the code in your plugin to see if I could identify how the plugin searches and returns citations but it is a little too complicated for my understanding of lua.

It seems strange because my config and set up is finding the correct bib file because it returns one of the citations but there must be something causing the scan of the `bib` file to not return all citations into the search. Does the `:Telescope bibtex` search the bib file each time the command is called? Does the plugin cache the `bib` file search?

I could provide a longer `bib` file if that would help?

Thanks for taking the time to help!

**Comment by noahares** [comment]
I don't think vimtex has anything to do with the issue. 

Yeah, the code has become a bit of a mess because of various features that were added over time. I can send you a `print` statement you can try when I am back at my desktop.

If you could provide the larger bib file, that would be great.

**Comment by noahares** [comment]
To see all labels that are read from your bibfile you can add `print(label)` after this line https://github.com/nvim-telescope/telescope-bibtex.nvim/blob/0b01f5c457f208ff2e12ce4dae89a61eec1ed69d/lua/telescope/_extensions/bibtex.lua#L119

**Comment by Rothbard-m** [comment]
Apologies for the delay in responding to you, I've been traveling and haven't had a chance to access my computer.

I added the print statement to your bibtex.lua file in the spot you listed, then called `:Telescope bibtex` but couldn't see any results from the print statement. I don't know where to view the results of this print statement in nvim, should it just be displaying in the messages/command line area? Do I need to call the `print(label)` function specifically or should it run the print as part of calling the `Telescope bibtex` command?

I've included my full bib file in an attachment. The `bib` filetype is not supported by GitHub in this chat area so I've saved it as a .txt file.

[eco.txt](https://github.com/nvim-telescope/telescope-bibtex.nvim/files/11483002/eco.txt)

**Comment by noahares** [comment]
You can see the output of the `print` by executing `:messages` inside neovim. That should list all the labels that were printed.
Thanks for the file. I will try it again with it.

**Comment by Rothbard-m** [comment]
Hi Noahares,

Thanks for the little tutorial on `:messages`. The results of the print statement seems to print all of my bib file. My inference from this investigation is that Telescope bibtex is reading the bib file as expected. See below results which you can see match up with the bib file.

I've attached a screenshot of the `:Telescope bibtex` function call too.

Output from print statement:
salop1982
leslie2020
chiang2005
edenhofer2013
byrne2022
meyer2022
schweppe1988
joskow2007
borenstein2005
joskow2006
helm2019
thompson2019
cramton2017
bohn1984
wolak2019
2009tappata
2000peltzman
1991bacon
1980varian
1985morgan
1993benabou
1983burdett
2004hastings
2008verlinda
2008yang
2015ritz
2021gugler
2023simeone
2021heim
2012cabral
2017duso


![image](https://github.com/nvim-telescope/telescope-bibtex.nvim/assets/95109526/59aa0088-08a7-4293-a94b-83b377d0ee35)

**Comment by noahares** [comment]
I see all entries in the picker when I use your bib-file. I have no clue what could be the problem on your machine :(

**Comment by Rothbard-m** [comment]
Ok thank you!

It must be something in Telescope is filtering my results after Telescope bibtex reads the bib file. I'll do some investigating with my base Telescope config and let you know if anything fixes the issue!

**Comment by Rothbard-m** [comment]
I did a lot more checking of my Telescope config and it turns out it was due to a file ignore pattern I had. I have the command line statistical software R installed and I have the string "R" listed in Telescope `file_ignore_patterns`. Once I removed this the full list of bib refs appears in the Telescope window.

Not sure how this file ignore pattern affects only some of the listed references. This is likely something I should have figured out without your help but I really appreciate the time to help.

Thanks for your help, the plugin is now working as expected. I consider this issue resolved.
